plugins/carbon_black_defense/komand_carbon_black_defense/actions/find_event/action.py

```diff
@@ -1,68 +1,79 @@
             input=FindEventInput(),
             output=FindEventOutput(),
         )
 
     def run(self, params={}):
-        host = self.connection.host
-        token = self.connection.token
-        connector = self.connection.connector
-        payload = dict()
-        for param in params:
-            if params[param]:
-                payload[param] = params[param]
-        self.logger.info(payload)
-        headers = {"X-Auth-Token": f"{token}/{connector}"}
-        url = host + FindEvent._URI
+        device_external_ip = params.get(Input.DEVICE_EXTERNAL_IP)
+        process_name = params.get(Input.PROCESS_NAME)
+        enriched_event_type = params.get(Input.ENRICHED_EVENT_TYPE)
+        process_hash = params.get(Input.PROCESS_HASH)
+        device_name = params.get(Input.DEVICE_NAME)
 
-        if payload:
-            result = requests.get(url, headers=headers, params=payload)
-        else:
-            result = requests.get(url, headers=headers)
-        try:
-            data = komand.helper.clean(result.json())
-        except ValueError:
-            self.logger.error(result.text)
-            raise Exception(
-                f"Error: Received an unexpected response"
-                f" (non-JSON or no response was received). Raw response in logs."
-            )
-        if result.status_code == 200:
-            return {
-                Output.SUCCESS: data["success"],
-                Output.LATESTTIME: data["latestTime"],
-                Output.RESULTS: data["results"],
-                Output.ELAPSED: data["elapsed"],
-                Output.MESSAGE: data["message"],
-                Output.TOTALRESULTS: data["totalResults"],
-            }
-        if result.status_code in range(400, 499):
-            raise Exception(
-                f"Carbon Black returned a {result.status_code} code."
-                f" Verify the token and host configuration in the connection. Response was: {result.text}"
-            )
-        if result.status_code in range(500, 599):
-            raise Exception(
-                f"Carbon Black returned a {result.status_code} code."
-                f" If the problem persists please contact support for help. Response was: {result.text}"
+        criteria = {}
+        exclusions = {}
+
+        if device_external_ip:
+            criteria["device_external_ip"] = device_external_ip
+        if process_name:
+            criteria["process_name"] = [process_name]
+        if enriched_event_type:
+            criteria["enriched_event_type"] = enriched_event_type
+        if process_hash:
+            criteria["process_hash"] = [process_hash]
+        if device_name:
+            criteria["device_name"] = [device_name]
+        if len(criteria) == 0:
+            raise PluginException(
+                cause="Error. Have not entered a criteria for action to run.",
+                assistance="Enter a criteria of at least one plugin input.",
```

Comment:
@pmara-r7 I think the single-process behavior is the same in the current version of the plugin. Since the update here is to switch to the latest API version, we should keep it as-is. In the future we could definitely enable multiple processes at once!

Reply:
Updated to properly wrap necessary inputs.

If we are looking at single inputs, should all []string inputs be converted to string?

Reply:
Responded in Slack @ablakley-r7

Reply:
Updated to allow multiple values in action input in plugin.spec.yaml file which the action passes directly.